src/tar.ts

```diff
@@ -5,78 +5,80 @@ import {touch} from './util'
 
 const debug = require('debug')('oclif-update')
 
-export async function extract(stream: NodeJS.ReadableStream, basename: string, output: string, sha?: string) {
-  const getTmp = () => `${output}.partial.${Math.random().toString().split('.')[1].slice(0, 5)}`
-  let tmp = getTmp()
-  if (fs.pathExistsSync(tmp)) tmp = getTmp()
-  debug(`extracting to ${tmp}`)
-  try {
-    await new Promise((resolve, reject) => {
-      const zlib = require('zlib')
-      const tar = require('tar-fs')
-      const crypto = require('crypto')
-      let shaValidated = false
-      let extracted = false
-      const check = () => shaValidated && extracted && resolve(null)
+export class Tar {
+  static async extract(stream: NodeJS.ReadableStream, basename: string, output: string, sha?: string) {
```

Comment:
Same as https://github.com/oclif/plugin-update/pull/325#discussion_r744994350. Please change back to a function.